Shopify/cli
# Stop setting previousAppId on `use` since `dev` already does

### WHY are these changes introduced?

On #2372 we started to set `previousAppId` on `use` to call the webhook accordingly. But as @Arkham noted, `dev` already sets the `previousAppId` no matter how the new one was chosen (either via `use` or `--config`). So I'm removing this logic from `use`. This also solves the issue of when the previous configuration doesn't exist anymore or is invalid.

### WHAT is this pull request doing?

Removes `previousAppId` from `use`'s `setAppInfo` and associated tests.

### How to test your changes?

Make sure the uninstall webhook is still sent when switching configs:

1. create an app
```bash
pnpm create-app --local --template=node --path ~/Desktop/multi-config-app
```
2. `link` it to more than one app
```bash
pnpm shopify app config link --path ~/Desktop/multi-config-app
```
3. run `dev`
```bash
pnpm shopify app dev --path ~/Desktop/multi-config-app
```
4. 👀 see the uninstall webhook being sent
5. run `dev` again with another `--config`
6. 👀 see the uninstall webhook being sent
7. switch to another config with `use`
```bash
pnpm shopify app config use --path ~/Desktop/multi-config-app
```
8. run `dev` again
9. 👀 see the uninstall webhook being sent

### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
